build_tools/docker/dockerfiles/samples.Dockerfile

```diff
@@ -26,16 +26,5 @@ RUN python3 -m pip install --ignore-installed \
     numpy \
     matplotlib \
     bottleneck \
-    tensorflow==2.8.0 \
-    tensorflow_hub==0.12.0
-
-# NOTE: 2022-05-27: protobuf 4.21.0, released on May 25, 2022 is incompatible
-# with prior releases. Specifically implicated are the above versions of
-# tensorflow, which seem to include it without a version pin and therefore
-# break out of the box. The next time the above versions are upgraded,
-# try removing this line and then, within the docker image, run:
-#   python3 -c "import tensorflow"
-# If that fails with a stack trace, put this line back.
-# On behalf of Google, we are sorry for the live at head philosophy
-# and shoddy version management leaking into everything. We're victims too.
-RUN python3 -m pip install protobuf==3.20.1 --force-reinstall
+    tensorflow==2.12.0 \
+    tensorflow_hub==0.13.0
```

Comment:
Also note: I'm _not_ updating the tf-nightly here [yet], which diverges from https://github.com/openxla/iree/blob/e4e23989319e5e231958ea3803c11d2710dc7730/integrations/tensorflow/test/requirements.txt#L1-L3

This is tracking what Colab uses. If we update the notebooks to install and use tf-nightly, _then_ I might also update the Dockerfile.